packages/components/src/presentation-components/properties/UniquePropertyValuesTargetSelector.tsx

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useState } from "react";
+import { ActionMeta, MultiValue } from "react-select";
+import { PropertyDescription, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
+import { PropertyFilterBuilderRuleValueProps } from "@itwin/components-react";
+import { IModelConnection } from "@itwin/core-frontend";
+import { ContentSpecificationTypes, Descriptor, DisplayValueGroup, FieldDescriptor, KeySet, Ruleset, RuleTypes, Value } from "@itwin/presentation-common";
+import { Presentation } from "@itwin/presentation-frontend";
+import { AsyncMultiTagSelect } from "../instance-filter-builder/MultiTagSelect";
+import { PRESENTATION_TARGETS_BATCH_SIZE } from "./UseNavigationPropertyTargetsLoader";
+import { translate } from "../common/Utils";
+
+/** @internal */
+export function UniquePropertyValuesTargetSelector(props: PropertyFilterBuilderRuleValueProps & { imodel: IModelConnection; descriptor: Descriptor }) {
+  const { imodel, descriptor, property, onChange, value } = props;
+  const [selectedTarget, setSelectedTarget] = useState<DisplayValueGroup[] | undefined>(() => getUniqueValueTargetFromProperty(value));
+  const [fieldDescriptor, setFieldDescriptor] = useState<FieldDescriptor | undefined>(() =>
+    descriptor.fields.find((field) => field.label === property.displayLabel)?.getFieldDescriptor(),
```

Comment:
Field should be looked up be the name

```suggestion
    descriptor.fields.find((field) => field.name === property.name)?.getFieldDescriptor(),
```